com.unity.render-pipelines.high-definition/Runtime/RenderPipeline/Camera/HDCamera.cs

```diff
@@ -1737,13 +1737,20 @@ Matrix4x4 GetJitteredProjectionMatrix(Matrix4x4 origProj)
         Matrix4x4 ComputePixelCoordToWorldSpaceViewDirectionMatrix(ViewConstants viewConstants, Vector4 resolution, float aspect = -1)
         {
             // In XR mode, use a more generic matrix to account for asymmetry in the projection
-            if (xr.enabled)
+            var useGenericMatrix = xr.enabled;
+
+            // Asymmetry is also possible from a user-provided projection, so we must check for it too.
+            // Note however, that in case of physical camera, the lens shift term is the only source of
+            // asymmetry, and this is accounted for in the optimized path below. Additionally, Unity C++ will
+            // automatically disable physical camera when the projection is overridden by user.
+            useGenericMatrix |= HDUtils.IsProjectionMatrixAsymmetric(viewConstants.projMatrix) && !camera.usePhysicalProperties;
+
+            if (useGenericMatrix)
             {
                 var transform = Matrix4x4.Scale(new Vector3(-1.0f, -1.0f, -1.0f)) * viewConstants.invViewProjMatrix;
                 transform = transform * Matrix4x4.Scale(new Vector3(1.0f, -1.0f, 1.0f));
                 transform = transform * Matrix4x4.Translate(new Vector3(-1.0f, -1.0f, 0.0f));
                 transform = transform * Matrix4x4.Scale(new Vector3(2.0f * resolution.z, 2.0f * resolution.w, 1.0f));
```

Comment:
In my PR, i simplified this to reduce the amount of matrix multiplication
I think it gives the same result but it might be worth rechecking
```
            var viewSpaceRasterTransform = new Matrix4x4(
                new Vector4(2.0f * resolution.z, 0.0f, 0.0f, -1.0f),
                new Vector4(0.0f, -2.0f * resolution.w, 0.0f, 1.0f),
                new Vector4(0.0f, 0.0f, 1.0f, 0.0f),
                new Vector4(0.0f, 0.0f, 0.0f, 1.0f));

            var transformT = viewConstants.invViewProjMatrix.transpose * Matrix4x4.Scale(new Vector3(-1.0f, -1.0f, -1.0f));
            return viewSpaceRasterTransform * transformT;
```

Reply:
Just added it back, I can confirm (at least locally) that it produces the exact same result